### PR TITLE
Improve references to HTTP errors

### PR DESCRIPTION
**Pull request by waldyrious**
Link all error references to the Errors page, and add external links in the Errors page for further details.

### CODE REVIEW

**Comment by waldyrious** [comment]
Marking as draft since this depends on #189.

File: _errors.md [hunk 1]
```diff
@@ -4,13 +4,13 @@ Uphold API uses the following error codes:
...
-416  | Requested Range Not Satisfiable
-429  | Too Many Requests -- Rate limit exceeded.
-500  | Internal Server Error -- Something went wrong in our server.
-503  | Service Unavailable -- We're temporarily offline for maintenance. Please try again in a little bit.
+400  | [Bad Request](https://tools.ietf.org/html/rfc7231#section-6.5.1) — Validation failed.
+401  | [Unauthorized](https://tools.ietf.org/html/rfc7235#section-3.1) — Bad credentials.
+403  | [Forbidden](https://tools.ietf.org/html/rfc7231#section-6.5.3) — Access forbidden.
+404  | [Not Found](https://tools.ietf.org/html/rfc7231#section-6.5.4) — Object not found.
+409  | [Conflict](https://tools.ietf.org/html/rfc7231#section-6.5.8) — Entity already exists.
+412  | [Precondition Failed](https://tools.ietf.org/html/rfc7232#section-4.2)
```
Comment (waldyrious) [inline _errors.md:12]: This is only mentioned in the docs regarding a malformed `Range` header (in the [Pagination section](https://github.com/uphold/docs/blob/master/_pagination.md)). Is it OK to add that description here, or do we use this error in other situation as well?

In any case, it seems odd for this to be the only error in this list without a custom description related to our API.

File: _authentication.md [hunk 2]
```diff
@@ -267,5 +267,5 @@ curl https://api.uphold.com/v0/me \
 
 You can use Basic Authentication by providing your email and password combination.
 
-If OTP (One-Time Password, also known as Two-Factor Authentication) is required, then you will get an HTTP 401 (Unauthorized) response, along with the HTTP header `OTP-Token: Required` and/or `OTP-Method-Id: Required`.
+If OTP (One-Time Password, also known as Two-Factor Authentication) is required, then you will get a [401 HTTP error](#errors), along with the HTTP header `OTP-Token: Required` and/or `OTP-Method-Id: Required`.
```
Comment (ruimarinho) [inline _authentication.md:270]: @waldyrious would be good to clarify which one do we show and under which scenario.

File: _errors.md (hunk 1 above)
Comment (ruimarinho) [reply _errors.md:12]: I don't think the description makes sense here as it is endpoint-specific. A 400 Bad Request may not always be a Validation failed error. 

We're missing a 416 here as well.

File: _authentication.md (hunk 2 above)
Comment (waldyrious) [reply _authentication.md:270]: Actually, both need to be passed together, so we should just replace `and/or` with `and` here. I'll do so in a bit.

File: _errors.md (hunk 1 above)
Comment (waldyrious) [reply _errors.md:12]: Good point. In fact, the descriptions we have here are essentially made obsolete by adding the links, which provide the canonical descriptions for these errors. What we have here is mostly redundant information, and in the cases it's not, it's inappropriately specific, as you point out. I'll go ahead and remove our custom descriptions.

> We're missing a 416 here as well.

Not sure what you mean — isn't that in the line immediately below?

**Review by hitmanmcc — APPROVED** [review]

File: _authentication.md (hunk 2 above)
Comment (waldyrious) [reply _authentication.md]: [Done](https://github.com/uphold/docs/compare/9e692daf5f8cb6545f978a21f478f947d13867cd..abf9dc5ec31154234ea42eb42040d3520c3af053#diff-abc260fb4e7cf8b00d11e36bbf570d59).

File: _errors.md (hunk 1 above)
Comment (ruimarinho) [reply _errors.md:12]: I think I've missed the first time, sorry!

**Pull request merged**